library/core/src/ffi/c_str.rs

```diff
@@ -755,6 +762,30 @@ impl AsRef<CStr> for CStr {
 #[unstable(feature = "cstr_internals", issue = "none")]
 #[cfg_attr(bootstrap, rustc_const_stable(feature = "const_cstr_from_ptr", since = "1.81.0"))]
 #[rustc_allow_const_fn_unstable(const_eval_select)]
+// Preconditions:
+// - ptr must be non-null and properly aligned
+// - ptr must point to a valid null-terminated string within isize::MAX bytes
+#[requires(!ptr.is_null() && {
```

Comment:
Once your other PR merges, can you please update this to use the new function?

Reply:
Sure will do